Errors from pdarrayclass.__del__ at end of unit test runs.

This is what I'm seeing after the unit tests run with `make test`:
```
 ---- Generated html report: file:///__w/arkouda/arkouda/.pytest/report.html ----
========= 3613 passed, 770 skipped, 156 warnings in 997.71s (0:16:37) ==========
Exception ignored in: <function pdarray.__del__ at 0x7fee6b1789a0>
Traceback (most recent call last):
  File "/__w/arkouda/arkouda/arkouda/numpy/pdarrayclass.py", line 548, in __del__
ImportError: sys.meta_path is None, Python is likely shutting down
Exception ignored in: <function pdarray.__del__ at 0x7fee6b1789a0>
Traceback (most recent call last):
  File "/__w/arkouda/arkouda/arkouda/numpy/pdarrayclass.py", line 548, in __del__
ImportError: sys.meta_path is None, Python is likely shutting down
Exception ignored in: <function pdarray.__del__ at 0x7fee6b1789a0>
Traceback (most recent call last):
  File "/__w/arkouda/arkouda/arkouda/numpy/pdarrayclass.py", line 548, in __del__
ImportError: sys.meta_path is None, Python is likely shutting down
Exception ignored in: <function pdarray.__del__ at 0x7fee6b1789a0>
Traceback (most recent call last):
  File "/__w/arkouda/arkouda/arkouda/numpy/pdarrayclass.py", line 548, in __del__
ImportError: sys.meta_path is None, Python is likely shutting down
make: *** [Makefile:790: coverage] Error 1
```

--- a/arkouda/numpy/pdarrayclass.py
+++ b/arkouda/numpy/pdarrayclass.py
@@ -542,13 +542,78 @@ class pdarray:
 
         self.registered_name: Optional[str] = None
 
-    def __del__(self):
+    def __del__(self):  # pragma: no cover
+        """
+        Attempt to delete the remote array from the Arkouda server when this
+        ``pdarray`` instance is garbage-collected.
+
+        This method is **best-effort only**: it performs a safe, silent cleanup
+        that never raises exceptions and skips work during interpreter shutdown.
+        The deletion request is sent via ``generic_msg("delete", {"name": name})``
+        if the Arkouda client is still available and the array has not already
+        been deleted.
+
+        Behavior summary
+        ----------------
+        1. Detects Python interpreter finalization and exits early if modules or
+           logging are being torn down.
+        2. Gracefully retrieves the array name; if missing, no deletion occurs.
+        3. Marks the object with a ``_deleted`` flag to prevent double deletion.
+        4. Logs the deletion attempt at DEBUG level, ignoring any logging errors.
+        5. Invokes the Arkouda ``delete`` command if possible, swallowing all
+           exceptions that might arise from network teardown, missing imports,
+           or client unavailability.
+
+        Notes
+        -----
+        - ``__del__`` is not guaranteed to run promptly or at all, depending on
+          reference cycles and interpreter state.
+        - For deterministic resource cleanup, users should call
+          :meth:`ak.pdarray.delete` explicitly instead of relying on the
+          destructor.
+        """
+        # 1) Bail out early during interpreter shutdown (imports/logging unsafe).
         try:
-            logger.debug(f"deleting pdarray with name {self.name}")
+            import sys
+
+            # If Python is finalizing or import machinery is gone, do nothing.
+            if (getattr(sys, "is_finalizing", None) and sys.is_finalizing()) or getattr(
+                sys, "meta_path", None
+            ) is None:
+                return
+        except Exception:
+            return
+
+        # 2) Get the array name safely; if absent, nothing to do.
+        try:
+            name = getattr(self, "name", None)
+            if not name:
+                return
+        except Exception:
+            return
+
+        # 3) Prevent re-entrancy / double delete on the same object.
+        try:
+            if getattr(self, "_deleted", False):
+                return
+            setattr(self, "_deleted", True)
+        except Exception:
+            # If we can't set the flag, still proceed best-effort.
+            pass
+
+        # 4) Best-effort log (don't let logging explode during teardown).
+        try:
+            logger.debug(f"deleting pdarray with name {name}")
+        except Exception:
+            pass
+
+        # 5) Send delete if the client is still importable/usable; swallow all errors.
+        try:
             from arkouda.client import generic_msg
 
-            generic_msg(cmd="delete", args={"name": self.name})
-        except (RuntimeError, AttributeError):
+            generic_msg(cmd="delete", args={"name": name})
+        except Exception:
+            # Channel closed, client gone, or shutdown in progress — ignore.
             pass
 
     def __bool__(self) -> builtins.bool:

